### PR TITLE
COMMERCE-9585 Integration tests for headless-commerce-admin-order module

### PR DESCRIPTION
**Pull request by smottal**
https://issues.liferay.com/browse/COMMERCE-9585

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ChannelResourceTest.java [hunk 1]
```diff
@@ -15,14 +15,236 @@
...
+				RandomTestUtil.randomLocaleStringMap(),
+				RandomTestUtil.randomLocaleStringMap(), true, 1, 1, 2022, 12, 0,
+				RandomTestUtil.nextInt(), 0, 0, 0, 0, 0, true, serviceContext);
+		COREntry corEntry = _corEntryLocalService.addCOREntry(
+			RandomTestUtil.randomString(), user.getUserId(), true,
+			RandomTestUtil.randomString(), 1, 1, 2022, 12, 0, 0, 0, 0, 0, 0,
+			true, RandomTestUtil.randomString(), RandomTestUtil.nextInt(),
+			COREntryConstants.TYPE_MINIMUM_ORDER_AMOUNT, null, serviceContext);
+
+		_commerceOrderTypeRel =
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ChannelResourceTest.java:94]: Move it before corEntry declaration

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderItemResourceTest.java [hunk 2]
```diff
@@ -14,15 +14,464 @@
...
+		_commerceCatalog = CommerceTestUtil.addCommerceCatalog(
+			testCompany.getCompanyId(), testGroup.getGroupId(),
+			_user.getUserId(), _commerceCurrency.getCode());
+
+		_commerceChannel = _commerceChannelLocalService.addCommerceChannel(
+			RandomTestUtil.randomString(), testGroup.getGroupId(),
+			RandomTestUtil.randomString(),
+			CommerceChannelConstants.CHANNEL_TYPE_SITE, null,
+			_commerceCurrency.getCode(), _serviceContext);
+		_commerceOrder = CommerceTestUtil.addB2BCommerceOrder(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderItemResourceTest.java:112]: Add an empty line before this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderNoteResourceTest.java [hunk 3]
```diff
@@ -14,15 +14,217 @@
...
+
+		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
+			_user.getUserId(), 0, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null,
+			RandomTestUtil.randomString() + "@liferay.com", null, null,
+			"business", 1,
+			ServiceContextTestUtil.getServiceContext(
+				testCompany.getCompanyId(), testGroup.getGroupId(),
+				_user.getUserId()));
+		CommerceCurrency commerceCurrency =
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderNoteResourceTest.java:63]: Add an empty line before this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleAccountGroupResourceTest.java [hunk 4]
```diff
@@ -58,53 +56,40 @@ public void setUp() throws Exception {
...
 
 	@Override
 	protected OrderRuleAccountGroup randomOrderRuleAccountGroup()
 		throws Exception {
 
-		CommerceAccountGroup commerceAccountGroup =
-			_commerceAccountGroupLocalService.addCommerceAccountGroup(
-				_user.getCompanyId(), RandomTestUtil.randomString(), 0, false,
-				null, _serviceContext);
+		AccountGroup accountGroup1 = _accountGroupLocalService.addAccountGroup(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleAccountGroupResourceTest.java:84]: Why the name is accountGroup1?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleOrderTypeResourceTest.java [hunk 5]
```diff
@@ -170,34 +149,26 @@ protected Long testGetOrderRuleIdOrderRuleOrderTypesPage_getId()
...
-	private OrderRuleOrderType _toOrderRuleOrderType(COREntryRel corEntryRel)
-		throws Exception {
-
 		CommerceOrderType commerceOrderType =
 			_commerceOrderTypeLocalService.getCommerceOrderType(
-				corEntryRel.getClassPK());
-		COREntry corEntry = _corEntryLocalService.fetchCOREntry(
-			corEntryRel.getCOREntryId());
+				orderRuleOrderType.getOrderTypeId());
+		COREntryRel corEntryRel = _corEntryRelLocalService.addCOREntryRel(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleOrderTypeResourceTest.java:162]: Add an empty line before this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderTypeResourceTest.java [hunk 6]
```diff
@@ -62,49 +61,28 @@ public void setUp() throws Exception {
...
+		_commerceTermEntry =
+			_commerceTermEntryLocalService.addCommerceTermEntry(
+				RandomTestUtil.randomString(), _user.getUserId(),
+				RandomTestUtil.randomBoolean(),
+				RandomTestUtil.randomLocaleStringMap(), 1, 1, 2022, 12, 0, 0, 0,
+				0, 0, 0, true, RandomTestUtil.randomLocaleStringMap(),
+				RandomTestUtil.randomString(), RandomTestUtil.nextDouble(),
+				RandomTestUtil.randomString(), StringPool.BLANK,
+				_serviceContext);
+		_corEntry = _corEntryLocalService.addCOREntry(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderTypeResourceTest.java:74]: Add an empty line before this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderTypeResourceTest.java [hunk 7]
```diff
@@ -251,63 +232,16 @@ private OrderType _addOrderType(OrderType orderType) throws Exception {
...
-			true, RandomTestUtil.randomString(), 0,
-			RandomTestUtil.randomString(), StringPool.BLANK, _serviceContext);
-
-		return _corEntryRelLocalService.addCOREntryRel(
+		_commerceTermEntryRel =
+			_commerceTermEntryRelLocalService.addCommerceTermEntryRel(
+				_user.getUserId(), CommerceOrderType.class.getName(),
+				commerceOrderType.getCommerceOrderTypeId(),
+				_commerceTermEntry.getCommerceTermEntryId());
+		_corEntryRel = _corEntryRelLocalService.addCOREntryRel(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderTypeResourceTest.java:240]: Add an empty line before this one

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ShippingAddressResourceTest.java [hunk 8]
```diff
@@ -14,16 +14,266 @@
...
+				false, RandomTestUtil.nextDouble(), true);
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.addCommerceChannel(
+				RandomTestUtil.randomString(), testGroup.getGroupId(),
+				RandomTestUtil.randomString(),
+				CommerceChannelConstants.CHANNEL_TYPE_SITE, null,
+				commerceCurrency.getCode(), serviceContext);
+
+		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ShippingAddressResourceTest.java:93]: Move it after serviceContext declaration

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ShippingAddressResourceTest.java [hunk 9]
```diff
@@ -14,16 +14,266 @@
...
+					commerceCurrency, commerceChannel, user, testGroup,
+					_commerceAccountLocalService.getCommerceAccount(
+						accountEntry.getAccountEntryId()),
+					commerceOrder),
+				serviceContext);
+
+		commerceOrderItem.setShippingAddressId(address.getAddressId());
+
+		_commerceOrderItemLocalService.updateCommerceOrderItem(
+			commerceOrderItem);
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/ShippingAddressResourceTest.java:150]: Don't we have a method for that?

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@smottal please check my comments, thanks!

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleAccountGroupResourceTest.java (hunk 4 above)
Comment (smottal) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-test/src/testIntegration/java/com/liferay/headless/commerce/admin/order/resource/v1_0/test/OrderRuleAccountGroupResourceTest.java]: I changed the way we create the OrderRuleAccountGroup

**Review by alessiorendina — APPROVED** [review]

**Pull request closed without merging**